src/test/run-pass/const-endianess.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(const_int_ops)]
+
+const BE_U32: u32 = 55u32.to_be();
+const LE_U32: u32 = 55u32.to_le();
+const BE_U128: u128 = 999999u128.to_be();
+const LE_I128: i128 = -999999i128.to_le();
+
+fn main() {
+    assert_eq!(BE_U32, 55u32.to_be());
```

Comment:
To ensure that the right hand side isn't const evaluated you need to pass the literal through a dummy function that isn't const fn and is also `#[inline(never)]`

Reply:
@oli-obk Doesn't the codebase have a dedicated black-boxing function for that purpose?

Reply:
It does, I think it's `test::black_box`, not sure if that's available in run-pass tests.

Reply:
Fixed. I was able to use `test::black_box`.

Btw. Do you think this test is correctly placed? I first put it under ui to just make sure it compiled, never ran any assertions. Then I moved it to run-pass :man_shrugging: 

Also, should I add tests for all methods I made `const fn`s?

Reply:
More tests are always great ;)

run-pass is the right place for it